documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,267 @@
+* `KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR`: the replication factor for the offsets topic. It sets the  
+`offsets.topic.replication.factor` property in the properties configuration file used by Kafka broker nodes on startup
+* `KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR`: the replication factor for the transaction topic. It sets the 
+`transaction.state.log.replication.factor` property in the properties configuration file used by Kafka broker nodes on startup
+* `kafka-storage`: a JSON string representing the storage configuration for the Kafka broker nodes. See related section
+* `zookeeper-storage`: a JSON string representing the storage configuration for the Zookeeper nodes. See related section
+* `kafka-metrics-config`: a JSON string representing the JMX exporter configuration for exposing metrics from Kafka broker nodes.
+ Removing this field means having no metrics exposed.
+* `zookeeper-metrics-config`: a JSON string representing the JMX exporter configuration for exposing metrics from Zookeeper nodes.
+ Removing this field means having no metrics exposed.
+ 
+An example of cluster configuration ConfigMap is the following.
+
+.Kafka cluster ConfigMap
+[source,yaml,options="nowrap"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-cluster
+  labels:
+    strimzi.io/kind: cluster
+    strimzi.io/type: kafka
+data:
+  kafka-nodes: "3"
+  kafka-image: "strimzi/kafka:latest"
+  kafka-healthcheck-delay: "15"
+  kafka-healthcheck-timeout: "5"
+  zookeeper-nodes: "1"
+  zookeeper-image: "strimzi/zookeeper:latest"
+  zookeeper-healthcheck-delay: "15"
+  zookeeper-healthcheck-timeout: "5"
+  KAFKA_DEFAULT_REPLICATION_FACTOR: "3"
+  KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: "3"
+  KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: "3"
+  kafka-storage: |-
+    { "type": "ephemeral" }
+  zookeeper-storage: |-
+    { "type": "ephemeral" }
+  kafka-metrics-config: |-
+    {
+      "lowercaseOutputName": true,
+      "rules": [
+          {
+            "pattern": "kafka.server<type=(.+), name=(.+)PerSec\\w*><>Count",
+            "name": "kafka_server_$1_$2_total"
+          },
+          {
+            "pattern": "kafka.server<type=(.+), name=(.+)PerSec\\w*, topic=(.+)><>Count",
+            "name": "kafka_server_$1_$2_total",
+            "labels":
+            {
+              "topic": "$3"
+            }
+          }
+      ]
+    }
+  zookeeper-metrics-config: |-
+    {
+      "lowercaseOutputName": true
+    }
+----
+
+==== Storage
+
+Kafka needs to save information like data logs (so received messages), consumer offsets and so on in some way; the same is true for 
+Zookeeper about znodes children and values.
+The Strimzi project allows to save such a data in an "ephemeral" way (using `emptyDir`) or in a "persistent-claim" way using persistent 
+volumes.
+It's possible to provide the storage configuration in the related ConfigMap using a JSON string as value for the 
+`kafka-storage` and `zookeeper-storage` fields.
```

Comment:
We should clarify here that storage cannot be changed. Only thing what matters is the configuration when creating the cluster.